lib/libvfio-user.c

```diff
@@ -462,186 +462,198 @@ handle_device_get_region_info(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
 }
 
 EXPORT int
-vfu_create_ioeventfd(vfu_ctx_t *vfu_ctx, size_t offset, uint32_t size, int fd, uint32_t flags, uint32_t index)
+vfu_create_ioeventfd(vfu_ctx_t *vfu_ctx, uint32_t region_idx, int fd,
+                     size_t offset, uint32_t size, uint32_t flags,
+                     uint64_t datamatch)
 {
+    vfu_reg_info_t *vfu_reg = &vfu_ctx->reg_info[region_idx];
 
     assert(vfu_ctx != NULL);
     assert(fd >= 0);
 
-    vfu_reg_info_t *vfu_reg = &vfu_ctx->reg_info[index];
+    vfu_reg = &vfu_ctx->reg_info[region_idx];
+
+    if (offset + size > vfu_reg->size) {
+        ERROR_INT(EINVAL);
+        return -1;
+    }
+
     ioeventfd_list_t *elem = malloc(sizeof(ioeventfd_list_t));
```

Comment:
nit, this makes it sounds like it's a list, but each element in as an ioeventfd, so shouldn't this be called "ioeventfd_t" ? Or given the name of the list, "subregion_t" ?